src/Backups/IBackupDataFileNameGenerator.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+namespace DB
+{
+
+struct BackupFileInfo;
+
+class IBackupDataFileNameGenerator
```

Comment:
Some comments would be helpful - for this class, for the `generate` function, for functions named `getBackupDataFileNameGenerator`.